test/functional/lua/uri_spec.lua

```diff
@@ -112,6 +112,13 @@ describe('URI methods', function()
         eq('C:\\xy\\åäö\\ɧ\\汉语\\↥\\🤦\\🦄\\å\\بِيَّ.txt', exec_lua(test_case))
       end)
     end)
+    describe('decode non-file uri', function()
+      it('uri_to_fname returns non-file uri unchanged', function()
+        eq('jdt://content/%5C/', exec_lua [[
+          return vim.uri_to_fname('jdt://content/%5C/')
+        ]])
+      end)
+    end)
```

Comment:
Would you add a test case that uri has not scheme?

Reply:
Yep, added one.